spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraConfCache.scala

```diff
@@ -0,0 +1,204 @@
+package org.apache.spark.sql.cassandra
+
+import com.datastax.spark.connector.cql.CassandraConnectorConf
+import com.datastax.spark.connector.rdd.ReadConf
+import com.datastax.spark.connector.writer.WriteConf
+import org.apache.commons.lang.StringUtils
+
+/**
+ * A local cache to store user defined Cassandra configuration settings. It includes
+ * per cluster level, per keyspace level and per table level settings. It also allows to
+ * update the settings. It's mainly used by Spark SQL, so user can define all kind of
+ * Cassandra configuration settings, and SQL query automatically get the settings to
+ * connect to, read from and write to Cassandra.
+ *
+ * Potentially it could be used by other components as well.
+ *
+ * If it's difficult to isolate the settings for your application, you still could pass
+ * explicitly those settings to the methods.
+ */
+
+class CassandraConfCache {
```

Comment:
Thanks for adding good ScalaDoc. But there is still an issue with the name. What you call a "cache" here is not really a cache, but a mutable object representing configuration/settings. You even call it differently (`confSettings`) at the use site. Caches are always secondary data sources. They only make data access to the primary data source faster, by either bringing data closer to the use-site or avoiding recomputation the same things over and over again. Evicting a real cache would only cause performance penalty. However, here, this object is used as a *primary* data source. If you removed the data from this object at random points in time, the information would be lost and the semantics of the program would change. Therefore it is not a cache and it should not be named cache.